app/src/main/java/com/gmail/mxwild/newhabit/movieslist/MoviesListViewModel.kt

```diff
@@ -31,7 +28,17 @@ class MoviesListViewModel : ViewModel() {
         }
     }
 
-    companion object {
-        const val TIME_DELAY: Long = 2_000
+    private suspend fun loadMoviesFromNetwork(): List<Movie> {
+        val topRatedMovies = NetworkService.movieApiService
+            .getTopRated()
+            .results
+            .map { it.id }
+
+        return topRatedMovies.distinct().map {
```

Comment:
Interesting - so you loaded movie already but then you call network again.

Reply:
No. We load movieId and than we load moviedetails by movieId.